cmd/juju/resource/charmresources.go

```diff
@@ -0,0 +1,196 @@
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/charm.v6-unstable"
+	charmresource "gopkg.in/juju/charm.v6-unstable/resource"
+	csparams "gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+
+	"github.com/juju/juju/charmstore"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// CharmResourcesCommand implements the "juju charm-resources" command.
+type CharmResourcesCommand struct {
+	baseCharmResourcesCommand
+}
+
+// NewCharmResourcesCommand returns a new command that lists resources defined
+// by a charm.
+func NewCharmResourcesCommand(resourceLister ResourceLister) modelcmd.ModelCommand {
+	var c CharmResourcesCommand
+	c.setResourceLister(resourceLister)
+	return modelcmd.Wrap(&c)
+}
+
+// Info implements cmd.Command.
+func (c *CharmResourcesCommand) Info() *cmd.Info {
+	i := c.baseInfo()
+	i.Name = "charm-resources"
+	return i
+}
+
+// SetFlags implements cmd.Command.
+func (c *CharmResourcesCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.setBaseFlags(f)
+}
+
+// Init implements cmd.Command.
+func (c *CharmResourcesCommand) Init(args []string) error {
+	return c.baseInit(args)
+}
+
+// Run implements cmd.Command.
+func (c *CharmResourcesCommand) Run(ctx *cmd.Context) error {
+	return c.baseRun(ctx)
+}
+
+// CharmResourceLister lists resources for the given charm ids.
+type ResourceLister interface {
+	ListResources(ids []charmstore.CharmID) ([][]charmresource.Resource, error)
+}
+
+type baseCharmResourcesCommand struct {
+	modelcmd.ModelCommandBase
+
+	// resourceLister is called by Run to list charm resources and
+	// uses juju/juju/charmstore.Client.
+	resourceLister ResourceLister
+
+	out     cmd.Output
+	channel string
+	charm   string
+}
+
+func (b *baseCharmResourcesCommand) setResourceLister(resourceLister ResourceLister) {
+	if resourceLister == nil {
+		resourceLister = b
+	}
+	b.resourceLister = resourceLister
+}
+
+func (c *baseCharmResourcesCommand) baseInfo() *cmd.Info {
+	return &cmd.Info{
+		Args:    "<charm>",
+		Purpose: "Display the resources for a charm in the charm store.",
+		Doc:     charmResourcesDoc,
+	}
+}
+
+func (c *baseCharmResourcesCommand) setBaseFlags(f *gnuflag.FlagSet) {
+	c.ModelCommandBase.SetFlags(f)
+	defaultFormat := "tabular"
+	c.out.AddFlags(f, defaultFormat, map[string]cmd.Formatter{
+		"tabular": FormatCharmTabular,
+		"yaml":    cmd.FormatYaml,
+		"json":    cmd.FormatJson,
+	})
+	f.StringVar(&c.channel, "channel", "stable", "the charmstore channel of the charm")
+}
+
+func (c *baseCharmResourcesCommand) baseInit(args []string) error {
+	if len(args) == 0 {
+		return errors.New("missing charm")
+	}
+	c.charm = args[0]
+
+	if err := cmd.CheckEmpty(args[1:]); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (c *baseCharmResourcesCommand) baseRun(ctx *cmd.Context) error {
+	// TODO(ericsnow) Adjust this to the charm store.
```

Comment:
I do not claim to know :) 
There are a few of puzzling to-dos in this area. I opted to leave them in case at some point in th future we'd come to realise what they mean... There must have been a really good reason to have these to-dos in first place, no?